hadoop-ozone/common/src/main/java/org/apache/hadoop/ozone/om/helpers/OmVolumeArgs.java

```diff
@@ -48,6 +48,19 @@
   private long quotaInNamespace;
   private long usedNamespace;
   private List<OzoneAcl> acls;
+  /**
+   * Reference count on this Ozone volume.
+   *
+   * When reference count is larger than zero, it indicates that at least one
+   * "lock" is held on the volume by some Ozone feature (e.g. multi-tenancy).
+   * Volume delete operation will be denied in this case, and user should be
+   * prompted to release the lock first via the interface provided by that
+   * feature.
+   *
+   * Volumes created using CLI, ObjectStore API or upgraded from older OM DB
+   * will have reference count set to zero by default.
+   */
+  private long refCount;
```

Comment:
This needs to be added to `getFromProtobuf`, `getProtobuf`, and `VolumeInfo` proto. If the value is not present, we can just set it to 0. Right now this does nothing, which indicates we need to add testing for existing volume create/delete requests and delete of a tenant volume, which would have caught this.

Reply:
Good catch.

tho btw the current broken impl still works, refCount is set directly through `omVolumeArgs.setRefCount(1L)` and persisted to the `VolumeTable`.